File: src/main/java/net/nodus/global/config/DeletedDataCleanupScheduler.java
package net.nodus.global.config;

import org.springframework.scheduling.annotation.Scheduled;
import org.springframework.stereotype.Component;

@Component
public class DeletedDataCleanupScheduler {

    private final DeletedDataCleanupService cleanupService;

    public DeletedDataCleanupScheduler(DeletedDataCleanupService cleanupService) {
        this.cleanupService = cleanupService;
    }

    @Scheduled(cron = "0 0 3 * * *")
    public void cleanup() {
        cleanupService.hardDeleteExpiredData();
    }
}

File: src/main/kotlin/net/nodus/config/DeletedDataCleanupScheduler.kt
package net.nodus.config

import org.springframework.scheduling.annotation.Scheduled
import org.springframework.stereotype.Component

@Component
class DeletedDataCleanupScheduler(
    private val cleanupService: DeletedDataCleanupService,
) {
    @Scheduled(cron = "0 0 3 * * *")
    fun cleanup() {
        cleanupService.hardDeleteExpiredData()
    }
}